modules/services/gotify-desktop.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.gotify-desktop;
+  tomlFormat = pkgs.formats.toml { };
+in
+{
+  meta.maintainers = with lib.hm.maintainers; [
+    joker9944
+  ];
+
+  options.services.gotify-desktop =
+    let
+      inherit (lib)
+        mkEnableOption
+        mkPackageOption
+        mkOption
+        literalExpression
+        types
+        ;
+    in
+    {
+      enable = mkEnableOption "Gotify daemon to receive messages and forward them as desktop notifications";
+
+      package = mkPackageOption pkgs "gotify-desktop" { };
+
+      url = mkOption {
+        type = types.str;
+        example = "wss://gotify.example.com:8443";
+        description = ''
+          Gotify server websocket URL, use wss:// prefix for TLS, or ws:// for unencrypted.
+        '';
+      };
+
+      token = mkOption {
+        type = types.oneOf [
+          types.str
+          (types.submodule {
+            options = {
+              command = mkOption {
+                type = types.str;
+              };
+            };
+          })
+        ];
+        example = literalExpression ''
+          "gotify-token"
+          or
+          { command = "''${lib.getExe pkgs.libsecret} lookup Title 'Gotify token'"; }
+          or
+          { command = "''${lib.getExe' pkgs.coreutils "cat"} /foo/bar/token"; }
+        '';
+        description = ''
+          Secret Gotify client token.
+          Either directly as string or as command for fetching client token.
+        '';
+      };
```

Comment:
These should just be documented options inside the `settings` option and use a submodule with `freeformType = tomlFormat.type` if these are just part of the generated `settings`.